src/dns-test.js

```diff
@@ -0,0 +1,317 @@
+/* global browser */
+const DNS_PACKET = require("dns-packet-dev");
+const { v4: uuidv4 } = require("uuid");
+
+const APEX_DOMAIN_NAME = "dnssec-experiment-moz.net";
+const SMIMEA_DOMAIN_NAME = "8c6976e5b5410415bde908bd4dee15dfb167a9c873fc4bb8a81f6f2a._smimecert.dnssec-experiment-moz.net";
+const HTTPS_DOMAIN_NAME = "_443._tcp.dnssec-experiment-moz.net";
+
+const RRTYPES = ['A', 'RRSIG', 'DNSKEY', 'SMIMEA', 'HTTPS', 'NEWONE', 'NEWTWO'];
+const RESOLVCONF_TIMEOUT = 5000; // 5 seconds
+const RESOLVCONF_ATTEMPTS = 2; // Number of UDP attempts per nameserver. We let TCP handle re-transmissions on its own.
+
+const TELEMETRY_TYPE = "dnssec-interference-study";
+const TELEMETRY_OPTIONS = {
+    addClientId: true,
+    addEnvironment: true
+};
+
+const MAX_TXID = 65535;
+const MIN_TXID = 0;
+
+var udp_query_proto;
+var tcp_query_proto;
+var measurement_id;
+
+var udpResponses = {"A":      {"data": "", "transmission": 0},
```

Comment:
(code style)
FYI: Keys in JavaScript objects don't need to be quoted. You can simply use `{A: { data: "", transmission: 0},`. In JSON (a data serialization format based on JavaScript), keys must be quoted though. But again, in JS quotes aren't required.
I'm mentioning this because I see inconsistent use of quotes in object literals throughout this patch.

Reply:
Addressed by f0f83c8436cd63d111d30de51f27fd2f90587279